### DISCUSSION TITLE
vercel tmp ( temp ) directory limits

### DISCUSSION THREAD

**Discussion by m0x61h0x64i** [vercel/vercel#8725] [Help] [answered]
what are the limits for using the tmp directory in hobby and pro vercel accounts ? i need to use it to cut random part of music in there and send it to client. so as you know i cant do it remotely i need the fs to do so.

**Comment by amyegan** [comment, marked as answer]
I believe the `/tmp` directory limit for Serverless Functions and both Hobby and Pro accounts is 512 MB

**Reply by m0x61h0x64i** [reply]
> I believe the `/tmp` directory limit for Serverless Functions and both Hobby and Pro accounts is 512 MB

can we upgrade it?

**Reply by amyegan** [reply]
I don't know of a way to upgrade Serverless Function `tmp` storage at this time. It is recommended that people use more permanent [file storage options](https://vercel.com/docs/concepts/solutions/file-storage#). 

With very large file processing, you would need to be careful about [Serverless Function execution timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limits as well.

**Reply by m0x61h0x64i** [reply]
> I don't know of a way to upgrade Serverless Function `tmp` storage at this time. It is recommended that people use more permanent [file storage options](https://vercel.com/docs/concepts/solutions/file-storage#).
> 
> With very large file processing, you would need to be careful about [Serverless Function execution timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limits as well.

yes, thats right, i know about s3 object cloud storage, but the problem is that project's music files should be in file system to cut random parts of it. so i need enough file system storage to do this and cant do this by musics links :(

**Reply by m0x61h0x64i** [reply]
> I don't know of a way to upgrade Serverless Function `tmp` storage at this time. It is recommended that people use more permanent [file storage options](https://vercel.com/docs/concepts/solutions/file-storage#).
> 
> With very large file processing, you would need to be careful about [Serverless Function execution timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limits as well.

let me know if there is a way to upgrade Serverless Function `tmp` storage. thanks.

**Reply by m0x61h0x64i** [reply]
> I don't know of a way to upgrade Serverless Function `tmp` storage at this time. It is recommended that people use more permanent [file storage options](https://vercel.com/docs/concepts/solutions/file-storage#).
> 
> With very large file processing, you would need to be careful about [Serverless Function execution timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limits as well.

did you get any news on finding a way to upgrade Serverless Function `tmp` storage ?

**Reply by m0x61h0x64i** [reply]
> I don't know of a way to upgrade Serverless Function `tmp` storage at this time. It is recommended that people use more permanent [file storage options](https://vercel.com/docs/concepts/solutions/file-storage#).
> 
> With very large file processing, you would need to be careful about [Serverless Function execution timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limits as well.

can we buy disk storage ?

**Reply by amyegan** [reply]
I have not found a way to upgrade or purchase additional Serverless Function `tmp` storage at this time

**Reply by m0x61h0x64i** [reply]
> I have not found a way to upgrade or purchase additional Serverless Function `tmp` storage at this time

okay thanks, what should i do now? should i message or email to someone or somewhere to ask for storage on vercel ?

**Reply by amyegan** [reply]
For now, you would need to use another service for [file storage](https://vercel.com/docs/concepts/solutions/file-storage) or a [database](https://vercel.com/docs/concepts/solutions/databases) to handle live data updates. I'll add increased `/tmp` storage to the requested features list.
